WorkflowUI/Sources/Hosting/WorkflowHostingController.swift

```diff
@@ -41,6 +41,8 @@ public final class WorkflowHostingController<ScreenType, Output>: WorkflowUIView
 
     private let (lifetime, token) = Lifetime.make()
 
+    private var lastEnvironmentAncestorPath: EnvironmentAncestorPath?
```

Comment:
I think we might have a potential retain cycle here, if this is storing strong references to the entire VC hierarchy above it. Can we weak-box each ancestor on the path, or does that break our ability to reliably compare paths?

Reply:
It's not a reference since it uses `ObjectIdentifier(node)` right? Eg, just a pointer.

Reply:
That's also my understanding.

Reply:
I talked through this with @watt and did some experimentation locally and it does look like `ObjectIdentifier` just keeps a reference to the pointer value, not the object itself.

One interesting note that @watt pointed out from [the documentation of `ObjectIdentifier`](https://github.com/apple/swift/blob/main/stdlib/public/core/ObjectIdentifier.swift#L15-L16):
> This unique identifier is only valid for comparisons during the lifetime of the instance.

Andrew was wondering if the pointer could be reused for a new instance and cause false positives on the equitability check.

This seems rare and very unlikely to occur, but we could add a weak box for these objects to better track the lifetime if we think it's worth the effort.

Reply:
I've updated this to use a weak box and `===` comparisons: fa85e0f